### ISSUE TITLE
Deleted alarm does not go away in HA

### ISSUE THREAD

**Issue by christianschneider89** [AaronDavidSchneider/SonosAlarm#12]
Hello

I deleted an alarm in Sonos, but the alarm is still there in HA.
Shouldn't it be deleted in HA too?

When toggling a nonexistent alarm, the log shows:
`
2020-12-17 22:51:37 ERROR (SyncWorker_4) [custom_components.sonos_alarm.switch] Home Assistant couldnt switch the alarm UPnP Error 800 received:  from 192.168.0.28
Traceback (most recent call last):
  File "/root/.homeassistant/custom_components/sonos_alarm/switch.py", line 213, in handle_switch_on_off
    self.alarm.save()
  File "/root/homeassistant/lib/python3.8/site-packages/pysonos/alarms.py", line 234, in save
    self.zone.alarmClock.UpdateAlarm(args)
  File "/root/homeassistant/lib/python3.8/site-packages/pysonos/services.py", line 210, in _dispatcher
    return self.send_command(action, *args, **kwargs)
  File "/root/homeassistant/lib/python3.8/site-packages/pysonos/services.py", line 499, in send_command
    self.handle_upnp_error(response.text)
  File "/root/homeassistant/lib/python3.8/site-packages/pysonos/services.py", line 553, in handle_upnp_error
    raise SoCoUPnPException(
pysonos.exceptions.SoCoUPnPException: UPnP Error 800 received:  from 192.168.0.28
`

Many thanks,
Chris

SONOS:
![image](https://user-images.githubusercontent.com/15085873/102547600-b84b4480-40b9-11eb-86fc-ea729fc0a6af.png)

HA:
![image](https://user-images.githubusercontent.com/15085873/102547566-abc6ec00-40b9-11eb-8108-316abc7e2fda.png)

**Comment by AaronDavidSchneider** [comment]
Thanks for the report!

Yes that's a good Point. I will have a look if that's possible.

Right now you would need to remove the integration and re-add it.

The same should apply for newly added alarms?!

**Comment by christianschneider89** [comment]
> The same should apply for newly added alarms?!

No, newly added alarms will appear automatically without restarting or re-adding the integration.

**Comment by AaronDavidSchneider** [comment]
Could you please test the PR #13 and comment wether this would fix the problem?

Just replace the `switch.py` file in your `custom_components/sonos_alarm` folder with the one from here:
https://raw.githubusercontent.com/AaronDavidSchneider/SonosAlarm/9d5ba3a0c32d1a917bf5cae047529e20c46dd124/custom_components/sonos_alarm/switch.py

Thanks!

**Comment by christianschneider89** [comment]
Thank you for the fast work!

I just tested it. When an alarm is deleted in the SONOS app, the entity in HA changes to "unavailable" and is not deleted. Is that the intention?
![image](https://user-images.githubusercontent.com/15085873/102596602-ab5e3d80-4119-11eb-95cc-a18e5cc504c3.png)

I just found another "bug".
When I change the room in the SONOS App in an alarm which does already exists, HA does not recognize it.
Only after restarting the server a new entity is created with the same ID and the _2 suffix:
![image](https://user-images.githubusercontent.com/15085873/102596614-b022f180-4119-11eb-95eb-6c0c91f35e29.png)

Thanks,
Chris

**Comment by AaronDavidSchneider** [comment]
Thanks again for your feedback! 

> I just tested it. When an alarm is deleted in the SONOS app, the entity in HA changes to "unavailable" and is not deleted. Is that the intention?

I have currently no idea if it is possible to delete the entry and device entirely, but that might be a cleaner solution.

Regarding the room change:
There should be an easy solution to this.

**Comment by AaronDavidSchneider** [comment]
Sorry for my inactivity. I plan to continue here once I have more time. I suppose these changes are anyway not really substantial to most users.

**Comment by AaronDavidSchneider** [comment]
Hi @christianschneider89,
I merged the PR now. And I will open up a new issue regarding the room change and work on it when I find some time.
Cheers
Aaron
